pkg/services/annotations/annotationsimpl/cleanup.go

```diff
@@ -2,25 +2,67 @@ package annotationsimpl
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"time"
 
 	"github.com/grafana/grafana/pkg/infra/db"
 	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/setting"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 // CleanupServiceImpl is responsible for cleaning old annotations.
 type CleanupServiceImpl struct {
-	store store
+	totalCleanedAnnotations       *prometheus.CounterVec
+	totalCleanedAnnotationsTags   prometheus.Counter
+	cleanupAnnotationsDuration    *prometheus.HistogramVec
+	cleanupAnnotationTagsDuration prometheus.Histogram
+	store                         store
 }
 
-func ProvideCleanupService(db db.DB, cfg *setting.Cfg) *CleanupServiceImpl {
-	return &CleanupServiceImpl{
+func ProvideCleanupService(db db.DB, cfg *setting.Cfg, promRegisterer prometheus.Registerer) (*CleanupServiceImpl, error) {
+	s := &CleanupServiceImpl{
+		totalCleanedAnnotations: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: "grafana",
+			Name:      "totals_cleaned_annotations",
```

Comment:
To follow https://prometheus.io/docs/practices/naming/ we should move `totals_` to the end as `_total`

```suggestion
			Name:      "cleaned_annotations_total",
```